src/mame/drivers/hp9845.cpp

```diff
@@ -722,6 +785,22 @@ WRITE_LINE_MEMBER(hp9845_base_state::t15_sts_w)
 	sts_w(T15_PA , state);
 }
 
+INPUT_CHANGED_MEMBER(hp9845_base_state::togglekey_changed)
+{
+	uintptr_t togglekey = (uintptr_t)param;
+	switch (togglekey) {
+	case 0:	// Shift lock
+		popmessage("SHIFT LOCK %s", m_io_shiftlock->read() ? "ON" : "OFF");
+		break;
+	case 1:	// Prt all
+		popmessage("PRT ALL %s", BIT(m_io_key0->read(), 1) ? "ON" : "OFF");
+		break;
+	case 2:	// Auto st
+		popmessage("AUTO ST %s", BIT(m_io_key0->read(), 17) ? "ON" : "OFF");
+		break;
+	}
```

Comment:
You should use the output system for this, not popmessage, and define an XML layout to display the keyboard LEDs on-screen.